modules/aws_asg/cloud-init.bats

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bats
+
+setup() {
+  export RUNDIR="${BATS_TMPDIR}/run"
+  export ETCDIR="${BATS_TMPDIR}/etc"
+  export MOCKLOG="${BATS_TMPDIR}/logs/mock.log"
+
+  mkdir -p \
+    "${RUNDIR}" \
+    "${ETCDIR}/rsyslog.d" \
+    "${ETCDIR}/default" \
+    "${BATS_TMPDIR}/bin" \
+    "${BATS_TMPDIR}/logs" \
+    "${BATS_TMPDIR}/returns"
+
+  rm -f "${MOCKLOG}"
+
+  touch "${ETCDIR}/hosts" "${ETCDIR}/hostname"
+
+  cat >"${ETCDIR}/default/travis-worker-bats" <<EOF
+export TRAVIS_TRAVIS_FAFAFAF=galaga
+export TRAVIS_TARVIS_SNOOK=fafa/___INSTANCE_ID___/faf
+EOF
+
+  cat >"${BATS_TMPDIR}/bin/mock" <<EOF
+#!/bin/bash
+echo "---> \$(basename \${0})" "\$@" >>${MOCKLOG}
+if [[ -f "${BATS_TMPDIR}/returns/\$(basename \${0})" ]]; then
+  cat "${BATS_TMPDIR}/returns/\$(basename \${0})"
+  exit 0
+fi
+echo "\${RANDOM}\${RANDOM}\${RANDOM}"
```

Comment:
I'm having the mocked-out executable echo a random string if there is no stub return value present, mostly so that every mocked-out executable *at least* has some unpredictable output so that I'm less likely to write brittle tests.